### PR TITLE
added mongo storage client

### PR DESCRIPTION
**Pull request by lambtron**
since this uses [`monk`](https://github.com/Automattic/monk), there are some known bugs that are commonly encountered if the node version is 4 or above.

Notably: 
https://github.com/Automattic/monk/issues/91
https://github.com/Automattic/monk/issues/114

I chose Monk for its simplicity, though Mongoose is probably the safer bet if people want to use newer node versions (i'm using 0.12 for this currently).

Just an FYI!

### CODE REVIEW

File: lib/mongo_storage.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+var db = require('monk');
+
+module.exports = function(path) {
```
Comment (RafaelCosman) [inline lib/mongo_storage.js]: Nice PR, @lambtron. In order to provide a consistent interface, this should be ```function(options)``` where options is an object with a path property.

Comment (lambtron) [inline lib/mongo_storage.js]: Oh shoot, good point ;)

I can make the update.

**Comment by audibleblink** [comment]
I love that Monk returns promises when you don't give it a callback. I hadn't heard of it, so I'm glad I saw this. I liked it so much that I went off and made [my own simple adapter](https://github.com/audibleblink/casbot2016/blob/develop/lib/mongo_storage.js) (I don't intend to hijack your PR, and I'm focusing on Harmony anyway.) My use-case needed more than the standard 3 stores.

**Comment by lambtron** [comment]
@audibleblink great to see you found use from Monk :)

File: lib/mongo_storage.js (hunk 1 above)
Comment (RafaelCosman) [inline lib/mongo_storage.js]: Thanks @lambtron! I'm reviewing and testing this now and we should have it merged soon. Thanks for your patience.

**Pull request merged**